internal/charmstore/migrations.go

```diff
@@ -181,7 +184,35 @@ func populateReadACL(db StoreDatabase) error {
 		if err := baseEntities.UpdateId(entity.URL, bson.D{{
 			"$set", bson.D{{"acls.read", readPerm}},
 		}}); err != nil {
-			return errgo.Notef(err, "cannot populate ACL for base entity %s", entity.URL)
+			return errgo.Notef(err, "cannot populate read ACL for base entity %s", entity.URL)
+		}
+		counter++
+	}
+	if err := iter.Close(); err != nil {
```

Comment:
won't this close conflict with the one above on line 176? It is now a double close.